barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -243,6 +243,43 @@ std::vector<FF> Execution::getDefaultPublicInputs()
     return public_inputs_vec;
 }
 
+/**
+ * @brief Run the bytecode, generate the corresponding execution trace and check the circuit for
+ *        execution of the supplied bytecode.
+ *
+ * @throws runtime_error exception when the bytecode is invalid.
+ */
+void Execution::check_circuit(AvmPublicInputs const& public_inputs, ExecutionHints const& execution_hints)
```

Comment:
There is already an env var AVM_FORCE_CHECK_CIRCUIT in Execution::prove which forces this. If we are not using it, maybe it's better to just change that IF for an IF-ELSE and not have to pollute the avm_prove interface?